### PR TITLE
add highlighting for hex & oct literals

### PR DESCRIPTION
**Pull request by jakecoffman**
Hope this is correct!

### CODE REVIEW

File: src/com/goide/GoAnnotator.java [hunk 1]
```diff
@@ -40,6 +40,9 @@ public void annotate(@NotNull PsiElement o, @NotNull AnnotationHolder holder) {
         o.putUserData(GoReference.IMPORT_USERS, ContainerUtil.<PsiElement>newArrayListWithCapacity(0));
       }
     }
+    else if (o instanceof GoLiteral) {
+      setHighlighting(o, holder, GoSyntaxHighlightingColors.NUMBER);
+    }
```
Comment (ignatov) [inline src/com/goide/GoAnnotator.java]: From the grammar (go.bnf):
```
Literal ::=
    int
  | float
  | floati
  | decimali
  | hex
  | oct
  | imaginary
  | rune
  | StringLiteral
  | char
```
So, you need to check that it's a hex or oct, but not a char or a rune.

**Comment by jakecoffman** [comment]
Is that better @ignatov?

**Comment by ignatov** [comment]
Thanks

**Comment by zolotov** [comment]
Hey, what about tests?;)

**Comment by jakecoffman** [comment]
I see GoHighlightingTest but can't figure out how to test for non-errors like hex highlighting. If you could give me a hint I will write some tests.

**Comment by zolotov** [comment]
@jakecoffman I've just pushed a simple test for that (GoHighlightingAnnotatorTest). TestData is placed at "testData/colorHighlighting".

**Pull request merged**